### PR TITLE
added relion v 2.1

### PR DESCRIPTION
**Pull request by JusticeForMikeBrown**
Thanks!

### CODE REVIEW

File: var/spack/repos/builtin/packages/relion/package.py [hunk 1]
```diff
@@ -32,8 +32,10 @@ class Relion(CMakePackage):
     electron cryo-microscopy (cryo-EM)."""
 
     homepage = "http://http://www2.mrc-lmb.cam.ac.uk/relion"
-    url      = "https://github.com/3dem/relion/archive/2.0.3.tar.gz"
+    url      = "https://github.com/3dem/relion"
 
+    version('2.1', git='https://github.com/3dem/relion.git',
+    commit='e7607a869687b636d3c39e0d5b6a9cba930fc3b2')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/relion/package.py]: Why this commit?

Comment (JusticeForMikeBrown) [reply var/spack/repos/builtin/packages/relion/package.py]: It's the latest.  In the past I was instructed to point the latest commit though I could rather set it to master if people prefer that instead.  This way we could always pull the latest fixes.

Comment (JusticeForMikeBrown) [reply var/spack/repos/builtin/packages/relion/package.py]: The entire program is beta though if this is such a big deal then I can change the version name.

Comment (junghans) [reply var/spack/repos/builtin/packages/relion/package.py]: @davydden, comments?

Comment (davydden) [reply var/spack/repos/builtin/packages/relion/package.py]: i would name it `develop` and point to the main branch.

File: var/spack/repos/builtin/packages/relion/package.py [hunk 2]
```diff
@@ -32,8 +32,10 @@ class Relion(CMakePackage):
     electron cryo-microscopy (cryo-EM)."""
 
     homepage = "http://http://www2.mrc-lmb.cam.ac.uk/relion"
-    url      = "https://github.com/3dem/relion/archive/2.0.3.tar.gz"
```
Comment (davydden) [inline var/spack/repos/builtin/packages/relion/package.py]: you sure `2.0.3` is still working with this change?

Comment (JusticeForMikeBrown) [reply var/spack/repos/builtin/packages/relion/package.py]: It does but it's no longer recommended to be used so I removed it as a release option

**Comment by JusticeForMikeBrown** [comment]
Relion only works with cuda 8 so I added that as a requirement.

File: var/spack/repos/builtin/packages/relion/package.py (hunk 2 above)
Comment (davydden) [reply var/spack/repos/builtin/packages/relion/package.py]: what I meant is that in the previous version, when `2.0.3` was around, I don't think Spack would be able to guess url to fetch it.

**Comment by tgamblin** [comment]
Not sure why the `tut` build test failed.  `r-rcpp` was fine.  I restarted it.

File: var/spack/repos/builtin/packages/relion/package.py (hunk 2 above)
Comment (JusticeForMikeBrown) [reply var/spack/repos/builtin/packages/relion/package.py]: I am going to add 2.0.3 back now.  2.1 is actually beta.

**Pull request merged**